testutil/simibc/network_link.go

```diff
@@ -0,0 +1,93 @@
+package simibc
+
+import channeltypes "github.com/cosmos/ibc-go/v3/modules/core/04-channel/types"
+
+// Ack represents and ack committed to block state
+type Ack struct {
+	Ack     []byte
+	Packet  channeltypes.Packet
+	Commits int
+}
+
+// Packet represents a packet committed to block state
+type Packet struct {
+	Packet  channeltypes.Packet
+	Commits int
+}
+
+// NetworkLink contains outboxes of packets and acknowledgements and
+// allows fine-grained control over delivery of acks and packets
+// to mimic a real relaying relationship between two chains.
+type NetworkLink struct {
```

Comment:
Awkward naming IMO. This isn't simulating a network link, but rather an IBC link, right? This name makes it sound like it is simulating an ethernet device or something.

Reply:
Hmm ok decent point.